publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java

```diff
@@ -198,21 +198,21 @@ public ResponseEntity generateAndPublish(@ApiParam(value = "message protocol", r
         catch (RemRemPublishException e) {
                 return new ResponseEntity(e.getMessage(), HttpStatus.NOT_FOUND);
         }
-        catch (Exception e) {
-            log.info("The result from REMReM Generate is not OK and have value: " + e.getMessage());
-            if (e.getMessage().startsWith(Integer.toString(HttpStatus.BAD_REQUEST.value()))) {
-                return new ResponseEntity(parser.parse(RemremPublishServiceConstants.GENERATE_BAD_REQUEST), HttpStatus.BAD_REQUEST);
-            } else if (e.getMessage().startsWith(Integer.toString(HttpStatus.SERVICE_UNAVAILABLE.value()))) {
-                return new ResponseEntity(parser.parse(RemremPublishServiceConstants.GENERATE_NO_SERVICE_ERROR), HttpStatus.SERVICE_UNAVAILABLE);
-            } else if (e.getMessage().startsWith(Integer.toString(HttpStatus.UNAUTHORIZED.value()))) {
-                return new ResponseEntity(parser.parse(RemremPublishServiceConstants.GENERATE_UNAUTHORIZED), HttpStatus.UNAUTHORIZED);
-            }else if (e.getMessage().startsWith(Integer.toString(HttpStatus.NOT_ACCEPTABLE.value()))) {
-                return new ResponseEntity(parser.parse(RemremPublishServiceConstants.ERLOOKUP_NONEFOUND), HttpStatus.NOT_ACCEPTABLE);
-            }else if (e.getMessage().startsWith(Integer.toString(HttpStatus.EXPECTATION_FAILED.value()))) {
-                return new ResponseEntity(parser.parse(RemremPublishServiceConstants.ERLOOKUP_MULTIPLEFOUND), HttpStatus.EXPECTATION_FAILED);
+        catch (HttpStatusCodeException e) {
+            log.info("The result from REMReM Generate is not OK and have value: " + e.getResponseBodyAsString());
+            if (e.getStatusCode().equals(HttpStatus.BAD_REQUEST)) {
+                return new ResponseEntity(parser.parse(e.getResponseBodyAsString()), HttpStatus.BAD_REQUEST);
+            } else if (e.getStatusCode().equals(HttpStatus.SERVICE_UNAVAILABLE)) {
+                return new ResponseEntity(parser.parse(e.getResponseBodyAsString()), HttpStatus.SERVICE_UNAVAILABLE);
+            } else if (e.getStatusCode().equals(HttpStatus.UNAUTHORIZED)) {
+                return new ResponseEntity(parser.parse(e.getResponseBodyAsString()), HttpStatus.UNAUTHORIZED);
+            }else if (e.getStatusCode().equals(HttpStatus.NOT_ACCEPTABLE)) {
+                return new ResponseEntity(parser.parse(e.getResponseBodyAsString()), HttpStatus.NOT_ACCEPTABLE);
+            }else if (e.getStatusCode().equals(HttpStatus.EXPECTATION_FAILED)) {
+                return new ResponseEntity(parser.parse(e.getResponseBodyAsString()), HttpStatus.EXPECTATION_FAILED);
             }
             else {
-                return new ResponseEntity(parser.parse(RemremPublishServiceConstants.GENERATE_INTERNAL_ERROR), HttpStatus.INTERNAL_SERVER_ERROR);
+                return new ResponseEntity(parser.parse(e.getResponseBodyAsString()), HttpStatus.INTERNAL_SERVER_ERROR);
             }
```

Comment:
While we're at it we could consider getting rid of the repetition altogether, e.g. with something like this:

```Java
HttpStatus result = HttpStatus.INTERNAL_SERVER_ERROR;
EnumSet<HttpStatus> passthrough = EnumSet.of(HttpStatus.BAD_REQUEST, HttpStatus.SERVICE_UNAVAILABLE, ...);
if (passthrough.contains(e.getStatusCode()) {
    result = e.getStatusCode();
}
return new ResponseEntity(parser.parse(e.getResponseBodyAsString()), result);
```